crates/sui-indexer/src/models/transactions.rs

```diff
@@ -77,10 +52,10 @@ pub fn commit_transactions(
     pg_pool_conn: &mut PgPoolConnection,
     tx_resps: Vec<SuiTransactionResponse>,
 ) -> Result<usize, IndexerError> {
-    let new_txn_iter = tx_resps.into_iter().map(NewTransaction::try_from);
+    let new_txn_iter = tx_resps.into_iter().map(|tx| tx.try_into());
 
     let mut errors = vec![];
-    let new_txns: Vec<NewTransaction> = new_txn_iter
+    let new_txns: Vec<Transaction> = new_txn_iter
```

Comment:
here the txn id will always be None?

Reply:
the id will be populated when reading back from the DB. We can just set it to null for new transactions

I am thinking we don't really need the id column, we have the transaction digest already to uniquely identify the rows already, adding a surrogate id won't give us any perf improvement.